applications/dirgraph_visualization/dirgraph_main.py

```diff
@@ -0,0 +1,619 @@
+import pysvzerod
+import pandas as pd
+import matplotlib.pyplot as plt
+import networkx as nx
+import json
+import dash
+from dash import html, dcc
+from dash.dependencies import Input, Output, State
+import plotly.graph_objects as go
+import os
+from dirgraph_utils import set_up_0d_network
+
+
+'''
+This file enables the visualization of 0D simulation results in a web app that displays the 0D network 
+as a directed graph. Users can interactively select nodes to view their parameters and simulation results.
+
+Simply provide the filepath for your simulation input JSON file and specify the directory where 
+you want to save the output directed graph.
+
+Created by Emilin Mathew (emilinm@stanford.edu) 
+'''
+
+
+def dirgraph(filepath, output_dir):
+    solver = pysvzerod.Solver(filepath)
+    solver.run()
+    results = pd.DataFrame(solver.get_full_result())
+    # If you want to export the simulation results as a csv
+    # results.to_csv('insert_file_name', sep=',', index=False, encoding='utf-8')
+    with open(filepath, 'r') as infile:
+        parameters = json.load(infile)
+
+    set_up_0d_network(
+        filepath,
+        name_type='id',  # Will take either 'id' or 'name',
+        draw_directed_graph=False,  # Enter false if you don't want to save the directed graph
+        output_dir=output_dir
+    )
+    base_name = filepath.rsplit('/', 1)[-1]
+    output_file = os.path.join(output_dir, os.path.splitext(base_name)[0] + "_directed_graph.dot")
+    G = nx.DiGraph(nx.nx_pydot.read_dot(output_file))
+    return results, parameters, G
+
+
+# Input filepath for 0d simulation & specify the output directory.
+results, parameters, G = dirgraph(
```

Comment:
Can this be changed to take in command line arguments instead of the user going in and editing this file?

Reply:
Yes, changed this and the description to match the change!